### ISSUE TITLE
[Bug] Unable to take offers that have a date expiry, even before offer expires.

### ISSUE THREAD

**Issue by AbandonedLand** [Chia-Network/chia-blockchain-gui#2386]
### What happened?

When using the (chia rpc wallet take_offer) on an offer with an expire date, I get an error.  If I use the GUI to accept the same offer I get conflicting messages.

![image](https://github.com/Chia-Network/chia-blockchain/assets/110296223/0b5c160f-91c7-48e7-8d63-e0ddfa8cbddc)

### Version

2.3.0

### What platform are you using?

Windows

### What ui mode are you using?

CLI

### Relevant log output

```shell
Request failed: RPC response failure: {"error": "This offer is no longer valid", "success": false, "traceback": "Traceback (most recent call last):\n  File \"chia\\rpc\\util.py\", line 49, in inner\n  File \"chia\\rpc\\util.py\", line 118, in rpc_endpoint\n  File \"chia\\rpc\\wallet_rpc_api.py\", line 2009, in take_offer\n  File \"chia\\wallet\\trade_manager.py\", line 782, in respond_to_offer\nValueError: This offer is no longer valid\n"}
```

**Comment by markelrod** [comment]
Transferred this enhancement request to the GUI repo.  Team discussed improving the messaging in the UI so that it is clear why the offer is no longer valid (accepted) and not only the fact that it expires in the future.

**Comment by wjblanke** [comment]
Should fixed in 2.4.0rc1
